From 319d56693de4aa66fde1630ea8a5b8777af7237d Mon Sep 17 00:00:00 2001
From: mkundaragi <mkundaragi@75f.io>
Date: Thu, 30 Jun 2022 06:41:56 +0000
Subject: [PATCH 28/35] Merged PR 4238: Restoring the snband point while
 restoring the point

Restoring the snband point while restoring the point

Related work items: #10702
---
 .../io/renatus/registration/ReplaceCCU.java   | 18 +++++--------
 .../a75f/io/api/haystack/RestoreCCUHsApi.java | 27 +++++++++++++++++++
 .../io/logic/bo/building/dab/DabEquip.java    |  3 +--
 .../io/logic/bo/building/vav/VavEquip.java    |  4 +--
 .../a75f/io/logic/ccu/restore/RestoreCCU.java |  5 ++++
 5 files changed, 41 insertions(+), 16 deletions(-)

diff --git a/app/src/main/java/a75f/io/renatus/registration/ReplaceCCU.java b/app/src/main/java/a75f/io/renatus/registration/ReplaceCCU.java
index 41047a62d..78fe63682 100644
--- a/app/src/main/java/a75f/io/renatus/registration/ReplaceCCU.java
+++ b/app/src/main/java/a75f/io/renatus/registration/ReplaceCCU.java
@@ -284,18 +284,7 @@ public class ReplaceCCU extends Fragment implements CCUSelect {
                         () -> {
                                 try {
                                     initRestoreCCUProcess(ccu);
-                                    List<HashMap> devices = CCUHsApi.getInstance().readAll("device and addr");
-                                    int pairingAddress = 1000;
-                                    for(HashMap device : devices){
-                                        int devicePairingAddress  = Integer.parseInt(device.get("addr").toString());
-                                        if(devicePairingAddress > pairingAddress){
-                                            pairingAddress = devicePairingAddress;
-                                        }
-                                    }
-                                    pairingAddress = (pairingAddress/100) * 100;
-                                    L.ccu().setSmartNodeAddressBand((short)pairingAddress);
-                                }
-                                catch(NullHGridException nullHGridException){
+                                } catch(NullHGridException nullHGridException){
                                     success.set(false);
                                     Log.i(TAG, nullHGridException.getMessage());
                                     nullHGridException.printStackTrace();
@@ -364,6 +353,7 @@ public class ReplaceCCU extends Fragment implements CCUSelect {
         restoreCCU.getCCUEquip(ccu.getCcuId());
         final int[] deviceCount = {restoreCCU.equipCountsInCCU(ccu.getCcuId(), ccu.getSiteCode())};
         restoreCCU.syncExistingSite(ccu.getSiteCode());
+        restoreCCU.getSettingPointsByCCUId(ccu.getCcuId());
         restoreCCU.getSystemProfileOfCCU(ccu.getCcuId(), ccu.getSiteCode());
         restoreCCU.getCMDeviceOfCCU(ccu.getCcuId(), ccu.getSiteCode());
         restoreCCU.getDiagEquipOfCCU(ccu.getCcuId(), ccu.getSiteCode());
@@ -381,7 +371,11 @@ public class ReplaceCCU extends Fragment implements CCUSelect {
         restoreCCU.getModbusSystemEquip(ccu.getCcuId(), ccu.getSiteCode(), deviceCount[0], equipResponseCallback);
         restoreCCU.getZoneEquipsOfCCU(ccu.getCcuId(), ccu.getSiteCode(), deviceCount[0], equipResponseCallback);
         restoreCCU.getOAOEquip(ccu.getCcuId(), ccu.getSiteCode(), deviceCount[0], equipResponseCallback);
+        HashMap pointMaps = CCUHsApi.getInstance().readEntity("point and snband");
+        int address = Integer.parseInt(pointMaps.get("val").toString());
+        L.ccu().setSmartNodeAddressBand((short)address);
         L.saveCCUState();
+
     }
 
     private void displayToastMessageOnRestoreSuccess(CCU ccu){
diff --git a/haystack/src/main/java/a75f/io/api/haystack/RestoreCCUHsApi.java b/haystack/src/main/java/a75f/io/api/haystack/RestoreCCUHsApi.java
index d71fc866f..3cf55f79f 100644
--- a/haystack/src/main/java/a75f/io/api/haystack/RestoreCCUHsApi.java
+++ b/haystack/src/main/java/a75f/io/api/haystack/RestoreCCUHsApi.java
@@ -766,6 +766,33 @@ public class RestoreCCUHsApi {
             }
         }
     }
+    public void restoreSNBandPointByCCUId(String ccuDeviceID){
+
+        HClient hClient = new HClient(CCUHsApi.getInstance().getHSUrl(), HayStackConstants.USER, HayStackConstants.PASS);
+        HDict settingPoints = new HDictBuilder().add("filter",
+                "setting and point and deviceRef == " + StringUtils.prependIfMissing(ccuDeviceID, "@")).toDict();
+        HGrid settingPointsGrid =  hClient.call("read", HGridBuilder.dictToGrid(settingPoints));
+
+        List<HashMap> pointMaps = ccuHsApi.HGridToList(settingPointsGrid);
+
+        pointMaps.forEach(pointMap -> {
+            Point pointDetails = new Point.Builder().setHashMap(pointMap).build();
+            String pointId = StringUtils.prependIfMissing(pointDetails.getId(), "@");
+            HashMap<Object, Object> point = ccuHsApi.readMapById(pointId);
+            if (point.isEmpty()) {
+                String pointLuid = ccuHsApi.addRemotePoint(pointDetails, pointDetails.getId().replace("@", ""));
+                ccuHsApi.setSynced(pointLuid);
+            } else {
+                CcuLog.i(TAG, "Point already imported "+pointDetails.getId());
+            }
+
+            Log.i(TAG, "restoreSNBandPointByCCUId: ");
+            SettingPoint.Builder sp = new SettingPoint.Builder().setHashMap(pointMaps.get(0));
+            sp.setVal(pointMaps.get(0).get("val").toString());
+            SettingPoint snBand = sp.build();
+            CCUHsApi.getInstance().updateSettingPoint(snBand, snBand.getId());
+        });
+    }
 
     public HGrid getDiagEquipByEquipId(String equipId){
         HClient hClient = new HClient(ccuHsApi.getHSUrl(), HayStackConstants.USER, HayStackConstants.PASS);
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/dab/DabEquip.java b/logic/src/main/java/a75f/io/logic/bo/building/dab/DabEquip.java
index c27767d26..5a661c43f 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/dab/DabEquip.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/dab/DabEquip.java
@@ -1045,8 +1045,7 @@ public class DabEquip
                 .setRoomRef(room)
                 .setFloorRef(floor).setHisInterpolate("cov").addMarker("analog1").addMarker("in")
                 .addMarker("damper").addMarker("dab")
-                .addMarker("temp").addMarker("sensor")
-                .addMarker("his").addMarker("cur").addMarker("logical").addMarker("zone")
+                .addMarker("sensor").addMarker("his").addMarker("logical").addMarker("zone")
                 .setGroup(String.valueOf(nodeAddr))
                 .setUnit("%")
                 .setTz(tz)
diff --git a/logic/src/main/java/a75f/io/logic/bo/building/vav/VavEquip.java b/logic/src/main/java/a75f/io/logic/bo/building/vav/VavEquip.java
index 9f5b63596..18741e937 100644
--- a/logic/src/main/java/a75f/io/logic/bo/building/vav/VavEquip.java
+++ b/logic/src/main/java/a75f/io/logic/bo/building/vav/VavEquip.java
@@ -1600,8 +1600,8 @@ public class VavEquip
                  .setRoomRef(room)
                  .setFloorRef(floor).setHisInterpolate("cov")
                  .addMarker("damper").addMarker("vav").addMarker(fanMarker).addMarker("zone")
-                 .addMarker("temp").addMarker("sensor").addMarker("analog1").addMarker("in")
-                 .addMarker("input").addMarker("his").addMarker("cur").addMarker("logical")
+                 .addMarker("sensor").addMarker("analog1").addMarker("in")
+                 .addMarker("input").addMarker("his").addMarker("logical")
                  .setGroup(String.valueOf(nodeAddr))
                  .setUnit("%")
                  .setTz(tz)
diff --git a/logic/src/main/java/a75f/io/logic/ccu/restore/RestoreCCU.java b/logic/src/main/java/a75f/io/logic/ccu/restore/RestoreCCU.java
index fac9c55d5..e33251974 100644
--- a/logic/src/main/java/a75f/io/logic/ccu/restore/RestoreCCU.java
+++ b/logic/src/main/java/a75f/io/logic/ccu/restore/RestoreCCU.java
@@ -394,4 +394,9 @@ public class RestoreCCU {
         Log.i(TAG, "Saving site details completed");
     }
 
+    public void getSettingPointsByCCUId(String ccuDeviceID){
+        Log.i(TAG, "Saving site details started");
+        restoreCCUHsApi.restoreSNBandPointByCCUId(ccuDeviceID);
+        Log.i(TAG, "Saving site details completed");
+    }
 }
-- 
2.31.0.windows.1

